### DISCUSSION TITLE
what replaces aiplatform.CustomContainerTrainingJobRunOp in google-cloud-pipeline-components V2

### DISCUSSION THREAD

**Discussion by raihan1khan** [kubeflow/pipelines#10684] [General]
this example currently doesn't work - https://github.com/GoogleCloudPlatform/fraudfinder/blob/main/vertex_ai/06_formalization.ipynb, it's using CustomContainerTrainingJobRunOp. Any idea

**Comment by raihan1khan** [comment]
@rimolive

**Reply by rimolive** [reply]
cc @connor-mccarthy 

It will be helpful if you can share the output of the block that's not working.

**Reply by raihan1khan** [reply]
so this block is not working with sdk v2
# custom training job component - script
    train_model_op = vertex_ai_components.CustomContainerTrainingJobRunOp(
        display_name=JOB_NAME,
        model_display_name=MODEL_NAME,
        container_uri=IMAGE_URI,
        staging_bucket=bucket_name,
        dataset=dataset_create_op.outputs["dataset"],
        base_output_dir=bucket_name,
        args=ARGS,
        replica_count=replica_count,
        machine_type=machine_type,
        training_fraction_split=train_split,
        validation_fraction_split=val_split,
        test_fraction_split=test_split,
        model_serving_container_image_uri=MODEL_SERVING_IMAGE_URI,
        project=project_id,
        location=region,
    ).after(dataset_create_op)

**Reply by giliolimirko** [reply]
CustomContainerTrainingJobRunOp has been removed in v2 and therefore it does not exists. You should run kfp=1.8

**Reply by raihan1khan** [reply]
@giliolimirko  - that's why i was asking what replaced custom container training job run op, if we are using 2.x then what's the solution. How can i use custom container training job

**Reply by rimolive** [reply]
I'd prefer to hear from the Google team to hear more about this. cc @chensun @zijianjoy @connor-mccarthy 

As far as I could see, and a quick research, says that there's no replacement for this component. Instead, you should begin using Containerized Python components.
